docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,411 @@ title: Data Grid - Editing
+```tsx
+<DataGrid editMode="row" columns={[{ field: 'name', editable: true }]} />
+```
 
-### Saving rows with nested structures
+Below there is a demo with row editing.
+To [start](#start-editing) and [stop](#stop-editing) editing use the same shortcuts from the cell editing (e.g. double-clicking a cell).
 
-Saving columns that make use of `valueGetter` can be done adding a `valueSetter`.
-The same [approach](/components/data-grid/editing/#saving-nested-structures) from the cell editing mode can be used here.
-Note that the `valueSetter` will be called for each field.
+{{"demo": "BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Server-side persistence [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+> ⚠ By design, when changing the value of a cell all `preProcessEditCellProps` callbacks from other columns are also called.
+> This happens to allow conditional validation (explained below) where the value of a cell impacts on the validation status of a another cell in the same row.
+> To only run validation if the value has changed check if the `hasChanged` param is `true`.
 
-If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.
+### Conditional validation
 
-To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.
+Having all cells of a row in edit mode allows validating a field based on the value of another one.
+To do that, start by adding the `preProcessEditCellProps` as explained in the [validation](#validation) section.
+When the callback is called, it will have an additional `otherFieldsProps` param contaning the props from the other fields in the same row.
+Use this param to check if the value from the current column is valid or not.
+Return normally the modified `props` containing the error.
+Once at the least one field has the `error` attribute equals to a truthy value the row will not leave the edit mode.
 
-You can then decide if you want to send the whole row or only the modified fields, by checking them against the previous row values.
+The following demo requires a value for the "Payment method" column only if the "Is paid?" column was checked.
 
-{{"demo": "RowEditServerSidePersistence.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+{{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
 
-### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+### Full-featured CRUD [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
 
-The following events can be imported and used to customize the row edition:
+The row editing makes possible to create a full-featured CRUD similar to those found in enterprise applications.
```

Comment:
I think it's worthwhile to spell out what CRUD stands for in case the reader hasn't encountered this acronym before.

```suggestion
Row editing makes it possible to create a full-featured CRUD (Create, Read, Update, Delete) app similar to those found in enterprise applications.
```

Reply:
I used "component" instead of "app".